hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/TestEndpointBase.java

```diff
@@ -95,12 +95,11 @@ public void testCustomMetadataHeadersSizeOverbig() {
       public void init() { }
     };
 
-    Exception exception = Assertions.assertThrows(
-            IllegalArgumentException.class,
-            () -> endpointBase.getCustomMetadataFromHeaders(s3requestHeaders));
-    Assert.assertEquals(
-            "Illegal user defined metadata. Combined size cannot exceed 2KB.",
-            exception.getMessage());
+    try {
+      endpointBase.getCustomMetadataFromHeaders(s3requestHeaders);
+    } catch (OS3Exception ex) {
+      Assert.assertTrue(ex.getCode().contains("MetadataTooLarge"));
+    }
```

Comment:
Here is a logic problem. If no exception is thrown, the UT will finish as normal but expected false.

Reply:
Thank you for catching that, I have fixed it now